### PR TITLE
Merge 1.7.6.x to develop

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Merge of bug fixes from `1.7.6.x` branch to `develop`
| Type?         | bug fix
| Category?     | ME
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Travis should be green

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15076)

### CODE REVIEW

File: src/Core/Routing/Exception/BuilderNotFoundException.php [hunk 1]
```diff
@@ -16,20 +16,22 @@
  *
  * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
  * versions in the future. If you wish to customize PrestaShop for your
- * needs please refer to https://www.prestashop.com for more information.
+ * needs please refer to http://www.prestashop.com for more information.
```
Comment (matks) [inline src/Core/Routing/Exception/BuilderNotFoundException.php:19]: Reverted update => bad

File: src/Core/Routing/EntityLinkBuilderInterface.php [hunk 2]
```diff
@@ -16,20 +16,43 @@
  *
  * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
  * versions in the future. If you wish to customize PrestaShop for your
- * needs please refer to https://www.prestashop.com for more information.
+ * needs please refer to http://www.prestashop.com for more information.
```
Comment (matks) [inline src/Core/Routing/EntityLinkBuilderInterface.php:19]: Bad git merge

File: src/Core/Routing/EntityLinkBuilderFactory.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
```
Comment (matks) [inline src/Core/Routing/EntityLinkBuilderFactory.php:19]: Needs to be updated

File: tests-legacy/PrestaShopBundle/Controller/ControllerTest.php [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<?php
+AdminManufacturersController<?php
```
Comment (matks) [inline tests-legacy/PrestaShopBundle/Controller/ControllerTest.php]: 😱

File: src/Core/Routing/EntityLinkBuilderInterface.php (hunk 2 above)
Comment (matks) [reply src/Core/Routing/EntityLinkBuilderInterface.php:19]: Actually it comes from 1.7.6.x files => licensing headers are outdated

File: src/Core/Routing/EntityLinkBuilderFactory.php (hunk 3 above)
Comment (matks) [reply src/Core/Routing/EntityLinkBuilderFactory.php:19]: Actually it comes from 1.7.6.x files => licensing headers are outdated

File: tests-legacy/PrestaShopBundle/Controller/ControllerTest.php (hunk 4 above)
Comment (matks) [reply tests-legacy/PrestaShopBundle/Controller/ControllerTest.php]: Fixed

**Review by matthieu-rolland — APPROVED** [review]
LGTM

**Comment by matks** [comment]
Thanks @matthieu-rolland for the review

**Comment by matks** [comment]
License headers to be fixed in https://github.com/PrestaShop/PrestaShop/pull/15092

**Pull request merged**